### PR TITLE
feat: converted all the tap tests with node:test

### PR DESCRIPTION
**Pull request by puskin**
1. Converted all the tests in the repo from `tap` to `node:test` and `node:assert`
2. Removed the `tap` dependency and related configuration files

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by Uzlopak — CHANGES_REQUESTED** [review]
Unfortunately not how we would like it. 

Lets take from surrogate.test.js the first test case:

```js
'use strict'

const test = require('tap').test
const validator = require('is-my-json-valid')
const build = require('..')

test('render a string with surrogate pairs as JSON:test 1', (t) => {
  t.plan(2)

  const schema = {
    title: 'surrogate',
    type: 'string'
  }

  const validate = validator(schema)
  const stringify = build(schema)
  const output = stringify('𝌆')

  t.equal(output, '"𝌆"')
  t.ok(validate(JSON.parse(output)), 'valid schema')
})
```

You did:

```js
'use strict'

const { describe } = require('node:test')
const { equal, ok } = require('node:assert')
const validator = require('is-my-json-valid')
const build = require('..')

describe('render a string with surrogate pairs as JSON:test 1', (t) => {
  const schema = {
    title: 'surrogate',
    type: 'string'
  }

  const validate = validator(schema)
  const stringify = build(schema)
  const output = stringify('𝌆')

  equal(output, '"𝌆"')
  ok(validate(JSON.parse(output)), 'valid schema')
})
```

what i expected:

```js
'use strict'

const { test } = require('node:test')
const validator = require('is-my-json-valid')
const build = require('..')

test('render a string with surrogate pairs as JSON:test 1', (t) => {
  t.plan(2)

  const schema = {
    title: 'surrogate',
    type: 'string'
  }

  const validate = validator(schema)
  const stringify = build(schema)
  const output = stringify('𝌆')

  t.assert.strictEqual(output, '"𝌆"')
  t.assert.ok(validate(JSON.parse(output)), 'valid schema')
})

```

Please revert and change accordingly. Thank you for your effort.

**Review by gurgunday — COMMENTED** [review]
Agree with

https://github.com/fastify/fast-json-stringify/pull/739#pullrequestreview-2308135470

**Review by Uzlopak — APPROVED** [review]
Can somebody another review. I just scrolled down in my phone and just checked if my remark was implemented.

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "lint": "standard",
     "lint:fix": "standard --fix",
     "test:typescript": "tsd",
-    "test:unit": "tap",
+    "test:unit": "node --test",
```
Comment (Eomm) [inline package.json:16]: We should add the coverage too:

```suggestion
    "test:unit": "c8 --100 node --test",
```

**Review by Eomm — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Uzlopak) [reply package.json:16]: also means adding c8 as devDependency

Comment (puskin94) [reply package.json:16]: @Eomm the `--100` flag is wishful thinking 😄 just added `c8`

**Review by Eomm — APPROVED** [review]

**Pull request merged**